Add getting started guide for maplibre gl native for android

michael-loc009 opened the pull request:
#808 
- Add the **getting-started-guide.md** for MapLibre GL Native for Android in **docs/mdbook/src/android**.
- Add the **MapLibre_Native_Android_Demo_Screenshot.png** in **docs/mdbook/src/assets**.

louwers commented:
@michael-loc009 Do you think Dokka is suitable for generic Markdown docs? Judging from having add pages and assets manually to the Gradle config, I think it was not intended that way. It also seems hard to add additional pages except for the main page.

Just asking for your thoughts. It's cool that you got it to work, but for the above reasons maybe it's better to just add it to `platform/android/docs/GETTING-STARTED.md` and set up an alternative docs site for Android (of course that is a separate issue).

louwers reviewed (changes requested):
Wr.t. the content of the guide, I think using `null` instead of `lateinit` is a Kotlin best practice, and using a layout inflater is current Android best practice. Maybe @ovivoda can verify.

I also think a screenshot without the buttons would be a bit cleaner.

michael-loc009 commented:
> @michael-loc009 Do you think Dokka is suitable for generic Markdown docs? Judging from having add pages and assets manually to the Gradle config, I think it was not intended that way. It also seems hard to add additional pages except for the main page.
> 
> Just asking for your thoughts. It's cool that you got it to work, but for the above reasons maybe it's better to just add it to `platform/android/docs/GETTING-STARTED.md` and set up an alternative docs site for Android (of course that is a separate issue).

After working with Dokka, I don't really recommend we should use the Dokka for generic Markdown docs since it requires some work with configurations. I agree with your suggestion on setting up an alternative docs site for Android which supports to add markdown documents conveniently.

michael-loc009 commented:
> Wr.t.



> Wr.t. the content of the guide, I think using `null` instead of `lateinit` is a Kotlin best practice, and using a layout inflater is current Android best practice. Maybe @ovivoda can verify.
> 
> I also think a screenshot without the buttons would be a bit cleaner.

I will update the PR based on your suggestion.

louwers commented:
Could you maybe add it to the Markdown book?

We have this set up already so it is the most convenient to use that for now.

michael-loc009 commented:
> lateinit



> Could you maybe add it to the Markdown book?
> 
> We have this set up already so it is the most convenient to use that for now.
> 
> https://maplibre.org/maplibre-gl-native/docs/book/

Should I remove the **GETTING-STARTED-GUIDE.md** in platform/android/MapboxGLAndroidSDK?

michael-loc009 commented:
Hi @louwers. I have made some updates for PR #939 based on our discussion:

1. I added the getting-started guide for MapLibre Native for Android in **/docs/mdbok/src/android**.
2. I updated the MapLibre_Native_Android_Demo_Screenshot with a new version without buttons.
3. I also made adjustments to the code example to init MapView in **MainActivity** using **lateinit** and **layout inflater**

michael-loc009 commented:
Hi @louwers. I removed the GETTING-STARTED-GUIDE.md and the example screenshot in platform/android/MapboxGLAndroidSDK. As for the dokka-generated document for MapboxGLAndroidSDK, I wonder if we need to add its reference into the mdbook as well.

ovivoda commented on docs/mdbook/src/android/getting-started-guide.md:
```diff
@@ -0,0 +1,113 @@
+                mapView?.onDestroy()
+            }
+        }
+        
```
let's also add 

`override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        mapView.onSaveInstanceState(outState)
    }`

michael-loc009 replied:
Okay @ovivoda . I added this part to the example in the getting started guide.

louwers commented on docs/mdbook/src/android/getting-started-guide.md:
````diff
@@ -0,0 +1,118 @@
+        }
+    ```
+
+2. Add the library as a dependency into your module Gradle file (usually `<project>/<app-module>/build.gradle`). Please replace `<version>` with a valid MapLibre GL version (e.g.: `org.maplibre.gl:android-sdk:10.0.2`).
````
```suggestion
2. Add the library as a dependency into your module Gradle file (usually `<project>/<app-module>/build.gradle`). Replace `<version>` with the latest MapLibre GL Native version (e.g.: `org.maplibre.gl:android-sdk:10.0.2`).
```

louwers replied:
Could you add a link to where you can find what the latest version is?

michael-loc009 replied:
I added the link that refers to the maven repository of MapLibre GL Native for Android so that  we can find the latest version in this commit 6385569ac80a64362250d25cd55cc29bbc604465.

louwers commented on docs/mdbook/src/android/getting-started-guide.md:
````diff
@@ -0,0 +1,118 @@
+
+4. Add a `MapView` to your layout XML file (usually `<project>/<app-module>/src/main/res/layout/activity_main.xml`).
+
+    ```xml
````
Could you remove the indentation in all code blocks so people more easily can copy paste?

michael-loc009 replied:
I removed the indentation in the code blocks in this commit eae556ea3b61954471a90396d88d11bb6dc18a66

louwers commented on docs/mdbook/src/android/getting-started-guide.md:
```diff
@@ -0,0 +1,118 @@
+
+6. Build and run the app. If you run the app successfully, a map will be displayed as seen in the screenshot below.
+<div style="text-align: center;">
+<img src="/assets/MapLibre_Native_Android_Demo_Screenshot.png" alt="Screenshot with the map in demotile style" width="50%" height="50%">
```
Please don't include screenshots in the repo. This will blow up the size of the repo over time. You can host them on GitHub. Drag and drop an image in the GitHub editor (anywhere) to get an URL.

michael-loc009 replied:
I removed the screenshot from the repos and hosted it directly via Github editor in these commits 7af2f822ec73c936f46ba079fff0d8da48680e78 and a938d5ceb4b9395aa1c71afb10c6f592f00345af.

louwers reviewed (changes requested):
In addition to my requested changes, please include a link to the Markdown book from the Android README. After that we can merge.

michael-loc009 commented on docs/mdbook/src/android/getting-started-guide.md:
```diff
@@ -0,0 +1,118 @@
+        class MainActivity : AppCompatActivity() {
+
+            // Declare a variable for MapView
+            private var mapView: MapView? = null
```
I replaced `null` with `lateinit` in this commit 74a2aa7313de730825fe72b59417815a8047f771

michael-loc009 commented:
> In addition to my requested changes, please include a link to the Markdown book from the Android README so that other people can find it. It is hosted here: https://maplibre.org/maplibre-gl-native/docs/book/
> 
> After that we can merge.

I updated the `Getting Started` section in android `README.md` by including the link that refers to getting started guide in mdbook in this commit 698c430266c2ea8224bdb2cfaf1466d8fd555711.

louwers commented on platform/android/README.md:
````diff
@@ -6,28 +6,7 @@ MapLibre GL Native for Android is a library for embedding interactive map views
-    ```
-
-3. Sync Gradle and rebuild your app.
+Visit [https://maplibre.org/maplibre-gl-native/docs/book/android/getting-started-guide.html](https://maplibre.org/maplibre-gl-native/docs/book/android/getting-started-guide.html) to view the `getting-started-guide`for MapLibre GL Native for Android.
````
```suggestion
Visit [https://maplibre.org/maplibre-gl-native/docs/book/android/getting-started-guide.html](https://maplibre.org/maplibre-gl-native/docs/book/android/getting-started-guide.html) to view the Getting Started Guide for MapLibre GL Native for Android.
```

louwers commented on docs/mdbook/src/android/getting-started-guide.md:
```diff
@@ -0,0 +1,113 @@
+
+        override fun onStart() {
+            super.onStart()
+            mapView?.onStart()
```
`mapView` cannot be `null`...

michael-loc009 replied:
I removed `?` from `mapView` in this commit d826413c683d9d396345171edf6e237351d211f7.

louwers reviewed (changes requested):
Final suggestion.

Since you figured out how to display Markdown on the main page of Dokka, could you add a short intro ("this is the MapLibre GL Native API for Android doccumentation") + a link to the MDBook? Then people opening the API documentation will be able to find the examples as well.

michael-loc009 commented:
> Final suggestion.
> 
> Since you figured out how to display Markdown on the main page of Dokka, could you add a short intro ("this is the MapLibre GL Native API for Android doccumentation") + a link to the MDBook? Then people opening the API documentation will be able to find the examples as well.

I added `README.md ` in `platform/android/MapboxGLAndroidSDK` to include the your suggested content in these commits 3383f32ae3ee04bef5e3aa561ab5de1af78bddd8 and  6bd73fb4749ec1bb032d8137e651fec341bcde50.
